selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/DashboardFactories.java

```diff
@@ -325,30 +308,98 @@ public void selectAction(AddAction addAction) {
+    loadPageTimeoutSec
+        .until(presenceOfElementLocated(By.xpath(AddActionWindow.PARAM_XPATH)))
         .clear();
     loader.waitOnClosed();
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.presenceOfElementLocated(By.xpath(AddActionWindow.PARAM_XPATH)))
+    loadPageTimeoutSec
+        .until(presenceOfElementLocated(By.xpath(AddActionWindow.PARAM_XPATH)))
         .sendKeys(paramValue);
     loader.waitOnClosed();
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(
-            ExpectedConditions.textToBePresentInElementValue(
-                By.xpath(AddActionWindow.PARAM_XPATH), paramValue));
+    loadPageTimeoutSec.until(
+        textToBePresentInElementValue(By.xpath(AddActionWindow.PARAM_XPATH), paramValue));
   }
 
   /** click 'Add' button on 'Add Action' window */
   public void clickAddOnAddAction() {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.elementToBeClickable(By.xpath(AddActionWindow.ADD_XPATH)))
-        .click();
+    loadPageTimeoutSec.until(elementToBeClickable(By.xpath(AddActionWindow.ADD_XPATH))).click();
   }
 
   /** scroll on 'Done' button */
   private void scrollOnDoneButton() {
     actionsFactory.createAction(seleniumWebDriver).moveToElement(doneBtn).perform();
     WaitUtils.sleepQuietly(2);
   }
+
+  public void waitFactoryName(String factoryName) {
+    loadPageTimeoutSec.until(
+        visibilityOfElementLocated(By.xpath(format(Locators.FACTORY_ITEM, factoryName))));
+  }
+
+  public void selectFactory(String factoryName) {
+    loadPageTimeoutSec
+        .until(visibilityOfElementLocated(By.xpath(format(Locators.FACTORY_NAME, factoryName))))
+        .click();
+  }
+
+  public String getFactoryRamLimit(String factoryName) {
+    return loadPageTimeoutSec
+        .until(
+            visibilityOfElementLocated(By.xpath(format(Locators.FACTORY_RAM_LIMIT, factoryName))))
+        .getText();
+  }
+
+  public void clickOnOpenInIdeButton(String factoryName) {
+    loadPageTimeoutSec
+        .until(visibilityOfElementLocated(By.xpath(format(Locators.FACTORY_ACTION, factoryName))))
+        .click();
+  }
+
+  public void waitFactoryNotExists(String factoryName) {
+    loadPageTimeoutSec.until(
+        invisibilityOfElementLocated(By.xpath(format(Locators.FACTORY_ITEM, factoryName))));
+  }
+
+  public void waitBulkCheckbox() {
+    loadPageTimeoutSec.until(visibilityOfElementLocated(By.xpath(Locators.BULK_CHECKBOX)));
+  }
+
+  public void selectAllFactoriesByBulk() {
+    loadPageTimeoutSec.until(visibilityOfElementLocated(By.xpath(Locators.BULK_CHECKBOX))).click();
+  }
+
+  public void clickOnDeleteFactoryBtn() {
+    loadPageTimeoutSec.until(visibilityOf(deleleFactoryButton)).click();
+  }
+
+  /** Click on the delete/remove button in the dialog window */
```

Comment:
Are there two possible names of the same button: delete or remove?

Reply:
Hm, only delete button.

Reply:
I see. In this case comment should be fixed a little: `delete/remove` > `delete`